var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -105,22 +105,22 @@ class PyTorch(PythonPackage, CudaPackage):
     cuda_arch_conflict = ('This version of Torch/Caffe2 only supports compute '
                           'capabilities ')
 
-    conflicts('cuda_arch=none', when='+cuda+caffe2',
+    conflicts('cuda_arch=none', when='+cuda',
               msg='Must specify CUDA compute capabilities of your GPU, see '
               'https://developer.nvidia.com/cuda-gpus')
-    conflicts('cuda_arch=52', when='@1.3.0:+cuda+caffe2',
+    conflicts('cuda_arch=52', when='@1.3.0:+cuda',
```

Comment:
What's the reason behind deprecating these versions? 
At least looks like `TORCH_CUDA_ARCH_LIST="3.5 5.2 6.0 6.1 7.0+PTX" ` is still in the official docker image.
https://github.com/pytorch/pytorch/blob/c8f665dcb626378ca9c51d7b70fc242524105441/Dockerfile#L47

Reply:
These conflicts were added by @glennpj in #14619, maybe he can explain where they came from. Hopefully I didn't misunderstand the code and the restriction only applies to `+cuda+caffe2`.

Reply:
Ping @glennpj. I was able to build PyTorch 1.4.0 with `cuda_arch=35` just fine (although I haven't tested it yet). Where did these conflicts come from?